kube-client/src/client/client_ext.rs

```diff
@@ -256,4 +408,61 @@ mod test {
 
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore = "needs cluster (will get svcs, clusterroles, pods, nodes)"]
+    async fn client_ext_fetch_ref_pods_svcs() -> Result<(), Box<dyn std::error::Error>> {
+        use k8s_openapi::api::{
+            core::v1::{Node, ObjectReference, Pod, Service},
+            rbac::v1::ClusterRole,
+        };
+
+        let client = Client::try_default().await?;
+        // namespaced fetch
+        let svc: Service = client
+            .fetch(&ObjectReference {
+                kind: Some(Service::kind(&()).into()),
+                api_version: Some(Service::api_version(&()).into()),
+                name: Some("kubernetes".into()),
+                namespace: Some("default".into()),
+                ..Default::default()
+            })
+            .await?;
+        assert_eq!(svc.name_unchecked(), "kubernetes");
+        // global fetch
+        let ca: ClusterRole = client
+            .fetch(&ObjectReference {
+                kind: Some(ClusterRole::kind(&()).into()),
+                api_version: Some(ClusterRole::api_version(&()).into()),
+                name: Some("cluster-admin".into()),
+                ..Default::default()
+            })
+            .await?;
```

Comment:
it would be good to see have an error test as well, to see what happens if people type inference the wrong thing (e.g. `:Service` with `ObjectRef` data for a `ClusterRole` say). might not be able to do something useful, but if we could reduce confusion, then that's less issues for us down the line.